_source/_posts/2020-01-21-angular-material-login.md

```diff
@@ -434,97 +399,133 @@ Open your browser and navigate to `http://localhost:4200`. Click on the **Play**
 You will be implementing a login form as part of your application. Before continuing, a few warnings. If you decide to run a production environment in which you host your own login form, make sure that you are using the secure HTTPS protocol, and you are hosting your site with a valid SSL certificate. Also, make sure to never store the login data in a session variable or the browser's storage. If you fail to take the necessary security precautions, your site may be vulnerable to attacks. As mentioned above, you will not be using any of the Angular-specific Okta libraries. Instead, install the `okta-auth-js` package by opening the terminal in the project's root folder and typing the following command.
 
 ```bash
-npm install -E @okta/okta-auth-js@4.8.0
+npm install -E @okta/okta-auth-js@5.9.1
 ```
 
+We'll add the configured `OktaAuth` library as a provider in Angular Dependency Injection system. Open `src/app/app.module.ts`. Add `import { OktaAuth } from '@okta/okta-auth-js';` to the file imports list at the top of the file.
+
+Next you'll add it to the `providers` array in the `@NgModule` configuration like the following code snippet.
+
+```ts
+@NgModule({
+  declarations: [
+    // ... component declarations. Leave as is
+  ],
+  imports: [
+    // ... imports array. Leave as is
+  ],
+  providers: [
+    {
+      provide: OktaAuth,
+      useValue: new OktaAuth({
+        issuer: 'https://{YourOktaDomain}/oauth2/default',
+        clientId: '{ClientId}',
+      })
+    }
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule {
+}
+```
+
+The `OktaAuth` object encapsulates the authentication, session management, and communication with the Okta servers. The `OktaAuth` constructor takes several options. The options provided here are the `issuer` and the `clientId`. In these options, you will have to replace `{YourOktaDomain}` with your Okta domain. The `{ClientId}` needs to be replaced with the client ID you obtained when you registered your application with Okta.
```

Comment:
We typically say lowercase the first letter: `yourOktaDomain` and `clientId`, but it doesn't matter as long as you're consistent in the post.

Reply:
This is pre-existing, but I changed all instances to camel case